config/config.go

```diff
@@ -10,37 +11,107 @@ import (
+	FileStoragePath    string `env:"FILE_STORAGE_PATH" json:"file_storage_path"`
+	DatabaseDSN        string `env:"DATABASE_DSN" json:"database_dsn"`
+	JWTSecretKey       string `env:"JWT_SECRET_KEY" json:"jwt_secret_key"`
+	ConcurrencyLimit   int    `env:"CONCURRENCY_LIMIT" json:"concurrency_limit"`
+	QueueSize          int    `env:"QUEUE_SIZE" json:"queue_size"`
+	EnableHTTPS        bool   `env:"ENABLE_HTTPS" json:"enable_https"`
+	CertFileName       string `env:"CERT_FILE_NAME" json:"cert_file_name"`
+	PrivateKeyFileName string `env:"PRIVATE_KEY_FILE_NAME" json:"private_key_file_name"`
+}
+
+func (c *Config) setDefaults() {
+	c.RunAddr = ":8080"
+	c.BaseURL = "http://localhost:8080"
+	c.ShortKeyLength = 8
+	c.LogLevel = "INFO"
+	c.FileStoragePath = fmt.Sprintf("%s/urls", os.TempDir())
+	c.DatabaseDSN = ""
+	c.JWTSecretKey = "a-string-secret-at-least-256-bits-long"
+	c.ConcurrencyLimit = 3
+	c.QueueSize = 0
+	c.EnableHTTPS = false
+	c.CertFileName = ""
+	c.PrivateKeyFileName = ""
 }
 
 // Initialize creates and initializes application configuration.
 func Initialize() (*Config, error) {
-	config := new(Config)
+	config := &Config{}
+	config.setDefaults()
+
+	configFile := parseFlagsForConfigFile()
 
-	flag.StringVar(&config.RunAddr, "a", ":8080", "address for app to run in format `host:port` or `:port`")
-	flag.StringVar(&config.BaseURL, "b", "http://localhost:8080", "base url which goes before short url id")
-	flag.IntVar(&config.ShortKeyLength, "u", 8, "short key length")
-	flag.StringVar(&config.LogLevel, "l", "INFO", "log level")
-	flag.StringVar(&config.FileStoragePath, "f", fmt.Sprintf("%s/urls", os.TempDir()), "path to file where to store urls")
-	flag.StringVar(&config.DatabaseDSN, "d", "", "string for connecting to postgres")
-	flag.StringVar(&config.JWTSecretKey, "j", "a-string-secret-at-least-256-bits-long", "key to sign jwt")
-	flag.IntVar(&config.ConcurrencyLimit, "cl", 3, "number of workers in url service worker pool")
-	flag.IntVar(&config.QueueSize, "q", 0, "worker pool jobs queue size, if not passed, will be set based on concurrency limit")
+	if configFile != "" {
+		if err := parseConfigFromFile(config, configFile); err != nil {
+			return nil, fmt.Errorf("failed to parse config from file: %w", err)
+		}
+	}
 
-	flag.Parse()
+	if err := parseCommandLineFlags(config); err != nil {
+		return nil, fmt.Errorf("failed to parse command line flags: %w", err)
+	}
 
-	err := env.Parse(config)
-	if err != nil {
-		return nil, err
+	if err := env.Parse(config); err != nil {
+		return nil, fmt.Errorf("failed to parse environment variables: %w", err)
 	}
 
 	return config, nil
 }
+
+// parseFlagsForConfigFile parses flags to get the config file path
+func parseFlagsForConfigFile() string {
+	var configFile string
+
+	tempFlagSet := flag.NewFlagSet("temp", flag.ContinueOnError)
+	tempFlagSet.StringVar(&configFile, "c", "", "config file name")
+
+	tempFlagSet.Parse(os.Args[1:])
+
+	if envConfigFile, set := os.LookupEnv("CONFIG"); set {
```

Comment:
В Go принято скорее так:

```suggestion
	if envConfigFile, ok := os.LookupEnv("CONFIG"); set {
```

Reply:
поправил